modules/xml/src/smithy4s/xml/Xml.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.xml
+
+import smithy4s.Blob
+import smithy4s.schema.Schema
+import fs2._
+import fs2.data.xml._
+import fs2.data.xml.dom._
+import cats.syntax.all._
+import cats.effect.Concurrent
+
+object Xml {
+
+  /**
+    * Reads an instance of `A` from a [[smithy4s.Blob]] holding an XML payload.
+    *
+    * Beware : using this method with a non-static schema (for instance, dynamically generated) may
+    * result in memory leaks.
+    */
+  def read[A: Schema](blob: Blob): Either[XmlDecodeError, A] =
+    readToStream[fs2.Fallible, A](blob)
+      .take(1)
+      .compile
+      .onlyOrError
+      .leftMap {
+        case x: XmlDecodeError => x
+        case other => new XmlDecodeError(XPath(List.empty), other.getMessage)
+      }
+
+  /**
+    * Writes the XML representation for an instance of `A` into a [[smithy4s.Blob]].
+    *
+    * Beware : using this method with a non-static schema (for instance, dynamically generated) may
+    * result in memory leaks.
+    */
+  def write[A: Schema](a: A): Blob = {
+    val result = writeToBytes[A](a).compile
+      .to(Collector.supportsArray(Array))
+
+    Blob(result)
+  }
+
+  /**
+    * Writes the XML representation for an instance of `A` into a String.
+    *
+    * Beware : using this method with a non-static schema (for instance, dynamically generated) may
+    * result in memory leaks.
+    */
+  def writeToString[A: Schema](a: A): Option[String] =
+    writeToStringStream[A](a).compile.last
+
+  type XmlByteStreamEncoder[F[_], A] =
+    smithy4s.codecs.Writer[Any, Stream[F, Byte], A]
+  type XmlByteStreamDecoder[F[_], A] =
+    smithy4s.codecs.Reader[F, Stream[F, Byte], A]
```

Comment:
cool, so I'm thinking actually that we need an `smithy4s-fs2` module, that could carry this : 

* a lot of what we do could actually be fs2-specific as opposed to http4s-specific, and that'd give us a chance to be more precise and avoid some coupling 
* the pagination work could go there 
* these types could move over-there, as there is nothing Xml specific to them